com.unity.multiplayer.mlapi/Runtime/Core/BatchingUtil.cs

```diff
@@ -0,0 +1,187 @@
+
+        public void PushLength(int length, ref PooledBitWriter writer)
+        {
+            // supports lengths up to 16-bit wide
+            if (length < 255)
+            {
+                writer.WriteByte((byte)length); // write the amounts of bytes that are coming up
+            }
+            else
+            {
+                writer.WriteByte(255); // mark larger size
+                writer.WriteByte((byte)(length % 256)); // write the length modulo 256
+                writer.WriteByte((byte)(length / 256)); // write the length divided by 256
+            }
+        }
+
+        public int PopLength(in BitStream messageStream)
+        {
+            int length = 0;
+            byte read = (byte)messageStream.ReadByte();
+            if (read != 255)
+            {
+                return read;
+            }
+            length += messageStream.ReadByte();
+            length += messageStream.ReadByte() * 256;
+
+            return length;
+        }
+
+        /// <summary>
+        /// GetTargetList
+        /// Returns the list of ClientId an item is targeted to
+        /// </summary>
+        /// <param name="item">the FrameQueueItem we want targets for</param>
+        private static List<ulong> GetTargetList(in FrameQueueItem item)
+        {
+            List<ulong> ret = new List<ulong>();
+            switch (item.QueueItemType)
+            {
+                case RPCQueueManager.QueueItemType.ServerRPC:
+                    ret.Add(item.NetworkId);
+                    break;
+                case RPCQueueManager.QueueItemType.ClientRPC:
+                    ret = item.ClientIds;
+                    break;
+                default:
+                    break;
+            }
+            return ret;
+        }
+
+        /// <summary>
+        /// QueueItem
+        /// Add a FrameQueueItem to be sent
+        /// </summary>queueItem
+        /// <param name="item">the threshold in bytes</param>
+        public void QueueItem(in FrameQueueItem item)
+        {
+            foreach (ulong clientId in GetTargetList(item))
+            {
+                // todo: actually queue and buffer. For now, the dict contains just one entry !!!
+
+                if (!SendDict.ContainsKey(clientId))
+                {
+                    SendDict[clientId] = new SendStream();
+                    SendDict[clientId].Item = item;
+                    SendDict[clientId].Writer = new PooledBitWriter(SendDict[clientId].Stream);
+
+                    SendDict[clientId].Writer.WriteBit(false); // Encrypted
+                    SendDict[clientId].Writer.WriteBit(false); // Authenticated
+
+                    switch (item.QueueItemType)
+                    {
+                        case RPCQueueManager.QueueItemType.ServerRPC:
+                            SendDict[clientId].Writer.WriteBits(MLAPIConstants.MLAPI_STD_SERVER_RPC, 6); // MessageType
```

Comment:
Magical '6'

Reply:
magical '6' is coming from existing `MessagePacker`. it's filling remaining 6 bits with `MLAPI_STD_SERVER_RPC` and first 2 bits are `WriteBit(false); // Encrypted` and `WriteBit(false); // Authenticated`

Reply:
ugh. It was already magical in NetworkedBehaviour `writer.WriteBits(MLAPIConstants.MLAPI_STD_SERVER_RPC, 6);`. But yeah, valid point. I'll add a comment.